src/main/java/net/minecraftforge/items/wrapper/EntityEquipmentInvWrapper.java

```diff
@@ -0,0 +1,170 @@
+package net.minecraftforge.items.wrapper;
+
+import com.google.common.collect.ImmutableList;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.inventory.EntityEquipmentSlot;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.IItemHandlerModifiable;
+import net.minecraftforge.items.ItemHandlerHelper;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Exposes the armor or hands inventory of an {@link EntityLivingBase} as an {@link IItemHandler} using {@link EntityLivingBase#getItemStackFromSlot} and
+ * {@link EntityLivingBase#setItemStackToSlot}.
+ */
+public abstract class EntityEquipmentInvWrapper implements IItemHandlerModifiable
+{
+    /**
+     * The entity.
+     */
+    protected final EntityLivingBase entity;
+
+    /**
+     * The slots exposed by this wrapper, with {@link EntityEquipmentSlot#index} as the index.
+     */
+    protected final List<EntityEquipmentSlot> slots;
+
+    /**
+     * @param entity   The entity.
+     * @param slotType The slot type to expose.
+     */
+    public EntityEquipmentInvWrapper(final EntityLivingBase entity, final EntityEquipmentSlot.Type slotType)
+    {
+        this.entity = entity;
+
+        final List<EntityEquipmentSlot> slots = new ArrayList<EntityEquipmentSlot>();
+
+        for (final EntityEquipmentSlot slot : EntityEquipmentSlot.values())
+        {
+            if (slot.getSlotType() == slotType)
+            {
+                slots.add(slot.getIndex(), slot);
+            }
+        }
```

Comment:
Unless someone adds a new `EntityEquipmentSlot` at runtime, this will never have holes. Vanilla uses the same values as the indexes of its item lists, so this guarantees that the `IItemHandler` uses the same indexes.

The enum constants *are* ordered by their index, so it would produce the same result.